### PR TITLE
[MERGE-THIS-FIRST] Load data into resume template and fix sync scheme

### LINKED ISSUE #81: Push sample resume and fill with data
Load a sample resume react component that gets filled with the values stored in database.

### LINKED ISSUE #99: Fix data update scheme
Currently, all the data is updated into local storage and the redux store is only updated once the PUT endpoint is called to update the data in the database, which happens every `n` (n = 10 for now) seconds. There are two drawbacks to this approach:

1. Local storage cannot serve as a state, so any update to local storage cannot be directed to the rendered resume, it can only pick up changes if the changes are applied to the redux state (called `resume`).
2. Local storage stores everything as strings, so all changes when committed to local storage require loading the currently stored resume, parsing it as a JSON, updating the object, and finally dumping it into a string, which is a super expensive operation. 

So, all the commits should be done to the redux state and local storage should be updated only when the PUT endpoint is called in order to store the newly found diff between the redux state and local storage.

### PR DESCRIPTION
**Pull request by frankhart2018**
Closes #81 
Closes #99 

**Implementation**

1. Implement a resume as a react component. 
2. Use the loaded resume from the redux state to fill in the react component as the data is filled in. 
3. Added reducers to update resume in the redux store.
4. Update sync time between redux store and local storage to 3 seconds.
5. Moved async sync function outside of `Basics` and into `LeftPanel`.
6. All immediate updates now go to the redux store, this is done by simply replacing the calls to update local storage to redux reducer dispatch. 
7. All entries in the sample resume now update in real-time. 

**Note**

- ~Do not merge this until #99 is merged, as without that change the UX is horrible.~
- I accidentally pushed changes for #99 here, so this now contains changes for both and can be merged.

### CODE REVIEW

**Comment by frankhart2018** [comment]
@dhairyaj Please also take a look at the CSS/Styling for this, fix those before merging this.

**Comment by frankhart2018** [comment]
@dhairyaj I have moved the styling issue for this to a new issue - #103. Please merge this after reviewing, this needs to be in master ASAP.

**Review by dhairyaj — APPROVED** [review]
LGTM

**Pull request merged**
